pkg/openstack/ironic.go

```diff
@@ -0,0 +1,75 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	ironicv1 "github.com/openstack-k8s-operators/ironic-operator/api/v1beta1"
+	corev1beta1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// ReconcileIronic -
+func ReconcileIronic(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
+	if !instance.Spec.Ironic.Enabled {
+		return ctrl.Result{}, nil
```

Comment:
The other option would be that we add handling for the Status update in this section before we return here. But imo, I prefer not even calling ReconcileIronic if the service is disabled. Which necessitates that we handle the status initialisation differently during the initial Reconcile function above.

Reply:
Done